### PR TITLE
feat: #6457 pipeline navigation refactor and updatable

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #6457

### CODE REVIEW

**Review by joshbalfour — APPROVED** [review]
LGTM 👍

File: packages/developer-portal/src/types/global.d.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export type Config = {
   swaggerUri: string
   PUSHER_KEY: string
   DEPLOYMENT_SERVICE_HOST: string
+  pipelineWhitelist?: string[]
```
Comment (willmcvay) [inline packages/developer-portal/src/types/global.d.ts:20]: Should be required field, you need to add it to the /core/index.tsx too where the config gets instantiated

**Review by willmcvay — CHANGES_REQUESTED** [review]
Just the config thing if you don't mind :-)

File: packages/developer-portal/src/types/global.d.ts (hunk 1 above)
Comment (bashleigh) [reply packages/developer-portal/src/types/global.d.ts:20]: oh ok, and force a default value of [] 👍🏼

Comment (willmcvay) [reply packages/developer-portal/src/types/global.d.ts]: Yep, nice one

**Review by willmcvay — APPROVED** [review]

**Pull request merged**